### PR TITLE
Change query to use id instead of timestamp

### LINKED ISSUE #578: Use IDs when fetching announcements
Yea so we should probably revert this back to using IDs, since I don't think timestamp will guarantee unique ordering (it might, but no reason to rely on the ambiguity there)

_Originally posted by @mds1 in https://github.com/ScopeLift/umbra-protocol/pull/558#discussion_r1273855710_

### PR DESCRIPTION
**Pull request by alexkeating**
Closes #578

### CODE REVIEW

**Review by mds1 — APPROVED** [review]

File: umbra-js/src/classes/Umbra.ts [hunk 1]
```diff
@@ -734,7 +734,7 @@ async function* recursiveGraphFetch(
   overrides?: GraphFilterOverride
 ): AsyncGenerator<any[]> {
   // retrieve the last ID we collected to use as the starting point for this query
-  const fromTimestamp = before.length ? (before[before.length - 1].timestamp as string | number) : false;
```
Comment (garyghayrat) [inline umbra-js/src/classes/Umbra.ts:737]: Fetching is ending up in some kind of a loop.

https://github.com/ScopeLift/umbra-protocol/assets/61768337/926aca5f-177d-4844-a2be-36a4f8513711

Comment (alexkeating) [reply umbra-js/src/classes/Umbra.ts:735]: https://github.com/ScopeLift/umbra-protocol/pull/594/commits/35983bb1b34f5f3dcaec573dd39a19a51fcd046e

Comment (garyghayrat) [reply umbra-js/src/classes/Umbra.ts:735]: https://github.com/ScopeLift/umbra-protocol/assets/61768337/4998c478-ad31-42d0-a2e4-43ae27a6f0fe

It's now not showing older txs that, on production, shows up after the second/full scan

Comment (garyghayrat) [reply umbra-js/src/classes/Umbra.ts:735]: > 

Couldn't reproduce this more than once. Can ignore it for now.

**Review by garyghayrat — APPROVED** [review]

**Pull request merged**